telegram/bot.py

```diff
@@ -302,68 +290,69 @@ def _validate_token(token: str) -> str:
 
         return token
 
-    @property  # type: ignore
-    @info
+    @property
+    def bot(self) -> User:
+        """:class:`telegram.User`: User instance for the bot as returned by :meth:`get_me`."""
+
+        if self._bot is None:
+            self._bot = self.get_me()
+        return self._bot
+
+    @property
     def id(self) -> int:
         """:obj:`int`: Unique identifier for this bot."""
 
-        return self.bot.id  # type: ignore
+        return self.bot.id
 
-    @property  # type: ignore
-    @info
+    @property
     def first_name(self) -> str:
         """:obj:`str`: Bot's first name."""
 
-        return self.bot.first_name  # type: ignore
+        return self.bot.first_name
 
-    @property  # type: ignore
-    @info
+    @property
     def last_name(self) -> str:
         """:obj:`str`: Optional. Bot's last name."""
 
         return self.bot.last_name  # type: ignore
```

Comment:
Do we still need those `# type: ignore` comments? 🤔 up to now that fact that `self.bot` was not `None` was hidden in the `@info` decorator, so I guess now mypy can do without them

Reply:
![IMG_20210116_234042_007](https://user-images.githubusercontent.com/58552395/104819650-e3f65180-5854-11eb-8922-cab76cc502d2.jpg)
Ig it's looks fine 🤔

Reply:
Umm.. i removed most `# type: ignore` comments but removing those 2 was causing mypy to fail in my local environment. So i left them as it was

Reply:
okay, let's leave them for now